### ISSUE TITLE
Zone Name for Condenser Placement in DX Coils is misusing the zone name as a node name

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#5174]
DX coil is incorrectly using the "Zone Name for Condenser Placement" as both the zone name _and_ the zone air node name in DXCoils::GetDXCoils  First occurrence is [here](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/DXCoils.cc#L1421-L1424); there are others.  

DXCoil( DXCoilNum ).SecZoneAirNodeNum is not really referencing the zone air node for the condenser zone.  It's creating a new air node using the zone name as the node name and registering it as NodeConnectionType_ZoneNode.  The node name is confusing, and the connection type is wrong, which trips the new error check in #5150 .

When the secondary condenser option (IsSecondaryDXCoilInZone) is active, SecZoneAirNodeNum  gets set to the current zone conditions [here](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/DXCoils.cc#L5442-L5447), so it appears that it's getting the right values.

If I'm following everything correctly, it seems that SecZoneAirNodeNum  could be dropped, and the existing Condenser Air Inlet Node Name (CondenserInletNodeNum) could be used instead.  The description for that node name would need to be updated and when it's registered it could be registered as NodeConnectionType_OutsideAirReference (for outdoor) and NodeConnectionType_Sensor when there is a IsSecondaryDXCoilInZone.  This would require moving up the check for IsSecondaryDXCoilInZone in the getinput so it's known or moving down the GetOnlySingleNode call for CondenserInletNodeNum.

**Comment by mjwitte** [comment]
Example file  HeatPumpSecondaryCoil can be a test file.

**Comment by Myoldmopar** [comment]
@mjwitte Are you, or anyone else, working on this yet?  I'm fine working on this one since it is blocking #5150.

**Comment by mjwitte** [comment]
I haven't gotten to this yet.  I can, but if you want to, that's cool too.

**Comment by Myoldmopar** [comment]
I shall, though a doctor visit interrupted today, so it will probably be
first thing Monday.

**Comment by Myoldmopar** [comment]
OK, so finally getting on this.  So this won't affect the SecZonePtr, which is required for registering the heat gain with SetupZoneInternalGain.  Fine.

Curious.  Right now the GetInput processing for A10 is:

```
if A10 is blank
  CondInletNodeNum(1) = 0;
else
  CondInletNodeNum(1) = GetAnOutdoorAirNode;
endif
```

You are saying we should leverage this CondInletNodeNum in lieu of the SecZoneAirNodeNum.  So then we need to hold before the GetAnOutdoorAirNode, because we may just call GetASensorNode.  I agree, but what happens if the user enters a value on A10 for an outdoor node, and also gives a value for the zone?  Do I need to add error handling for that case as well?

@mjwitte @rraustad Thoughts?

**Comment by mjwitte** [comment]
@Myoldmopar Good question.  Ideally, all of the OutdoorDryBulb, OutdoorHumRat, etc. variables should be renamed to something like CondInletDryBulb, etc.  And if the CondInletNodeNum is pointing to the correct node then many of the if ( DXCoil( DXCoilNum ).IsSecondaryDXCoilInZone ) blocks could go away.  

As far as the input is concerned, perhaps the condenser inlet node name should be ignored and the secondary zone air node could be used?  Then [this block](https://github.com/NREL/EnergyPlus/blob/ddcd0d2256264e2813015baad01e388d7ec9fb8e/src/EnergyPlus/DXCoils.cc#L5681-L5686) that uses ZT could go away because the zone air node should hold the desired conditions.

But if you want to go for a minimal change, then yes, you'll need to check to make sure that node name is not an outdoor air node.

**Comment by rraustad** [comment]
Prior to allowing the DX coil condenser to connect to a zone, the NodeConnectionType_ZoneNode was only used in a couple places, to register the zone node and to register the zone node name for plenum objects. The new calls in DXCoil using GetOnlySingleNode looks to me to only be a method to find a zone index number, nothing more. So these could be changed to use NodeConnectionType_Sensor or something else.

I'm not sure why the condenser wasn't simply connected to the zone via zone inlet and exhaust nodes. I think that would have made more sense. The example file just disregards the condenser air inlet node name input field in the coil object.

```
Coil:Cooling:DX:SingleSpeed,
 ,                        !- Condenser Air Inlet Node Name
 NORTH ZONE;              !- Zone Name for Condenser Placement
```

**Comment by rraustad** [comment]
@Myoldmopar @mjwitte @JasonGlazer 

Several possible changes come to mind:
1) change this line to register the zone node as an internal node (sensor node doesn't work). This will allow #5150 to pass. This one is working and gives the same answer. I did make another change here. If the secondary coil is in a controlled zone, the code uses that zones air node instead of creating a new node (as "Zone Name Air Node"). I still have to test this using a controlled zone for the secondary coil to make sure that part works.
2) actually use the zone air node name instead and register as sensor node. Is this change transitiionable? Same issues with 1) for controlled vs uncontrolled zones. If uncontrolled, a zone air node name is not available.
3) deprecate the Zone Name for Condenser Placement and instead use the Condenser Air Inlet Node Name field.This option would take more time to correct and test. This option would connect the secondary coil to a zone inlet and exhaust node. Not sure of the implications of this before testing and validation. Would also require a condenser outlet air node field in Coil:Cooling:DX and condenser inlet and outlet in Coil:Heating:DX.

```
DXCoil( DXCoilNum ).SecZoneAirNodeNum = GetOnlySingleNode( Alphas( 18 ), ErrorsFound, CurrentModuleObject, cAlphaFields( 18 ), NodeType_Air, NodeConnectionType_ZoneNode, 1, ObjectIsNotParent );
DXCoil( DXCoilNum ).SecZonePtr = FindItemInList( Alphas( 18 ), Zone );

Coil:Cooling:DX:SingleSpeed,
 ,                        !- Condenser Air Inlet Node Name
 NORTH ZONE;              !- Zone Name for Condenser Placement

Coil:Heating:DX:SingleSpeed,
 ,                        !- Evaporator Air Inlet Node Name
 NORTH ZONE,              !- Zone Name for Evaporator Placement
```

**Comment by rraustad** [comment]
I could probably get rid of the SecZoneAirNodeNum and just use zone conditions based on SecZonePtr using ZT(SecZonePtr) and ZoneHumRat(SecZonePtr)

**Comment by rraustad** [comment]
Got rid of SecZoneAirNodeNum, no need to register that node now.

**Comment by rraustad** [comment]
PR https://github.com/NREL/EnergyPlus/pull/5339
PT#109121630

**Comment by rraustad** [comment]
Results comparison

![5174_compare](https://cloud.githubusercontent.com/assets/5778156/11459305/a0b382a0-96a1-11e5-9590-f67435b0802d.png)

**Comment by mjwitte** [comment]
Closed via #5339
